pkg/runtime/framework/plugins/mpi/mpi_test.go

```diff
@@ -807,6 +803,63 @@ func TestValidate(t *testing.T) {
 				).
 				Obj(),
 		},
+		"runtime does not have Launcher but TrainJob has only 1 numNodes": {
+			info: runtime.NewInfo(
+				runtime.WithMLPolicySource(utiltesting.MakeMLPolicyWrapper().
+					WithMLPolicySource(*utiltesting.MakeMLPolicySourceWrapper().
+						MPIPolicy(ptr.To[int32](1), ptr.To(trainer.MPIImplementationOpenMPI), nil, ptr.To(true)).
+						Obj(),
+					).
+					Obj(),
+				),
+			),
+			newObj: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "test").Trainer(&trainer.Trainer{NumNodes: ptr.To(int32(1))}).Obj(),
+		},
+		"runtime does not have Launcher even though MPI with runLauncherAsNode has 2 numNodes": {
+			info: runtime.NewInfo(
+				runtime.WithMLPolicySource(utiltesting.MakeMLPolicyWrapper().
+					WithMLPolicySource(*utiltesting.MakeMLPolicySourceWrapper().
+						MPIPolicy(ptr.To[int32](1), ptr.To(trainer.MPIImplementationOpenMPI), nil, ptr.To(true)).
+						Obj(),
+					).
```

Comment:
Adding "Node" by using https://github.com/kubeflow/trainer/blob/3ef89f4f79501390036ea1677588d351c5f9a8c4/pkg/runtime/runtime.go#L138

You can learn how to add "Node" and "Launcher" by `WithPodSet` function in torch_test.go.